src/CMakeLists.txt

```diff
@@ -303,6 +307,7 @@ list(APPEND rover_libs
   control
   commands
   utils
+  autonomous
```

Comment:
Instead of linking `rover_libs` to `autonomous`, should we be linking `mission_control_interface` to `autonomous`? Since we're getting rid of the usages in `Rover.cpp`, that should be the only user I think.